src/server/common/page/model.js

```diff
@@ -0,0 +1,10 @@
+/**
+ * @template Payload
+ * @template Data
+ * @typedef {{
+ *   clean: (payload: Payload) => Data,
+ *   validate: (data: Data) => ValidationResult
+ * }} Model<Payload, Data>
+ */
+
+/** @import {ValidationResult} from './validation.js' */
```

Comment:
I didn't mean to add this - but this is give or take where I see the direction of travel.